src/Swarm/TUI/Controller.hs

```diff
@@ -837,32 +848,31 @@ makeEntity s e = do
           s ^? gameState . robotMap . at 0 . _Just . robotContext . defStore
 
   case isActive <$> (s ^. gameState . robotMap . at 0) of
-    Just False ->
-      continue $
-        s
-          & gameState . replStatus .~ REPLWorking mkTy Nothing
-          & gameState . robotMap . ix 0 . machine .~ initMachine mkPT empty topStore
-          & gameState %~ execState (activateRobot 0)
-    _ -> continueWithoutRedraw s
+    Just False -> do
+      gameState . replStatus .= REPLWorking mkTy Nothing
+      gameState . robotMap . ix 0 . machine .= initMachine mkPT empty topStore
+      gameState %= execState (activateRobot 0)
```

Comment:
Would this be better as `zoom gameState $ activateRobot 0`?  (That would ultimately have the same effect, right?)

Reply:
That does not work, because the `fused-effect` signature does not find the right instance.

It is not that much boiler plate, so I can probably live with it for now.